binventory/binventory/binventory.m

```diff
@@ -22,52 +22,21 @@ - (void)removeStatusLabelForType:(int)arg1;
 @implementation DKTrashTile
 
 - (void)dk_updateCount {
-    NSUInteger x = 0;
+    long x = 0;
     
-    for (NSURL *url in Trashes) {
-        FSRef	ref;
-        CFURLGetFSRef((CFURLRef)url, &ref);
-        FSCatalogInfo	catInfo;
-
-        OSErr	err	= FSGetCatalogInfo(&ref, kFSCatInfoValence, &catInfo, NULL, NULL, NULL);
-        if (err == noErr)
-            x += catInfo.valence;
-        
+    for (NSURL *url in Trashes)
+    {
+        if ([[NSFileManager defaultManager] fileExistsAtPath:[NSString stringWithFormat:@"%@/.DS_Store", url.path]])
+            x -= 1;
+        x += [[[NSFileManager defaultManager] contentsOfDirectoryAtPath:[url path] error:nil] count];
```

Comment:
This method is slower and not as accurate as the FS API which is why I didn't use it here. It's easier/faster to get the valence of the directory inode.